hw_diag/utilities/event_streamer.py

```diff
@@ -125,9 +127,10 @@ def enqueue_event(self, event) -> None:
 
     def process_queued_events(self) -> None:
```

Comment:
this is non blocking queue processor can be called from any thread. A separate thread for processing events might make it simpler.